nomad/jobspec/parse.go

```diff
@@ -1603,6 +1603,7 @@ func parseUpdate(result **api.UpdateStrategy, list *ast.ObjectList) error {
 		"healthy_deadline",
 		"progress_deadline",
 		"auto_revert",
+		"auto_promote",
```

Comment:
the `api.UpdateStrategy` object that this attempting to parse into doesn't have an `AutoPromote` field, because the dependency version for the Nomad API needs to be updated.